modules/system/assets/ui/js/select.js

```diff
@@ -87,23 +87,25 @@
                         return $request
                     },
                     processResults: function (data, params) {
-                        var results = data.result;
-                        var options = [];
-
-                        for (var i in results) {
-                            if (results.hasOwnProperty(i)) {
-                                var isObject = i != null && i.constructor.name === 'Object';
-                                
-                                options.push({
-                                    id: isObject ? results[i].id : i,
-                                    text: isObject ? results[i].text : results[i],
-                                });
-                            };
-                        };
-
-                        return {
-                            results: options,
-                        };
+                        var results = data.result || data.results,
+                            options = []
+
+                        if (results[0] && results[0].id) { // Already in Select2 format
+                            options = results
+                        }
+                        else {
+                            for (var i in results) {
+                                if (results.hasOwnProperty(i)) {
+                                    options.push({
+                                        id: i,
+                                        text: results[i],
+                                    })
+                                }
+                            }
+                        }
+
+                        data.results = options
+                        return data
```

Comment:
@jimcottrell Would you be able to keep the original `return` call as is, and add any extra parameters (such as `pagination`) as required? It might seem risky sending the entire data object as is to the Select2.

Reply:
Can you clarify what the risk would be?

Reply:
@jimcottrell I suppose I'm just being overly paranoid and considering that, potentially in the future, Select2 may introduce other applicable properties to their data format that may have unintended issues with our implementation. I just figured if we only looked for and passed through the pertinent keys (`results` and `pagination`) and discard the rest, we just keep the data that's expected.

Reply:
I can see what you mean, although actually I took the other side of that, because I wrote it this way so that if the Select2 format did add any other functionality requiring other keys, they'd still be passed through without requiring changes to `processResults`.

I can change it if you think it's important, but from my perspective, the data came from an AJAX handler specifically written to send search results to Select2, and Select2 is a library we included to begin with, so the security needs should be minimal.

Reply:
@jimcottrell You make good points - we'll leave it as is for now.